### PR TITLE
VpnGateway Feature: Get-AzVirtualNetworkGatewayVpnclientConnectionHealth

### PR DESCRIPTION
**Pull request by ninweninwe**
## Description
This is a new command to get Virtual Network Gateway Vpn Client Health.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by ninweninwe** [comment]
Here is azure-powershell-cmdlet-review-pr link.
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/303

**Comment by MikhailTryakhov** [comment]
@ninweninwe if it won't be fixed today (conflicting files) the PR would be closed :(

**Comment by ninweninwe** [comment]
> @ninweninwe if it won't be fixed today (conflicting files) the PR would be closed :(

I am using the local SDK.  The new SDK is not there.  What do I need to do ?

**Comment by niander** [comment]
@ninweninwe 
There is an static analysis error:
`Cmdlet 'Get-AzVirtualNetworkGatewayVpnClientConnectionHealth' has multiple parameter sets, but no defined default parameter set.`
Please take a look into this. Make sure you run static analysis build step before committing your changes.

**Comment by ninweninwe** [comment]
> @ninweninwe
> There is an static analysis error:
> Cmdlet 'Get-AzVirtualNetworkGatewayVpnClientConnectionHealth' has multiple parameter sets, but no defined default parameter set.
> Please take a look into this. Make sure you run static analysis build step before committing your changes.

How to run run static analysis build step locally?

**Comment by ninweninwe** [comment]
> @ninweninwe
> There is an static analysis error:
> Cmdlet 'Get-AzVirtualNetworkGatewayVpnClientConnectionHealth' has multiple parameter sets, but no defined default parameter set.
> Please take a look into this. Make sure you run static analysis build step before committing your changes.

My swagger was checked in.  How to import the updated azure sdk for net?

**Comment by ninweninwe** [comment]
> @ninweninwe
> There is an static analysis error:
> Cmdlet 'Get-AzVirtualNetworkGatewayVpnClientConnectionHealth' has multiple parameter sets, but no defined default parameter set.
> Please take a look into this. Make sure you run static analysis build step before committing your changes.
> 
> My swagger was checked in.  How to import the updated azure sdk for net?

error MSB3027: Could not copy "C:\
Users\ninwe\.nuget\packages\microsoft.azure.management.network\20.0.0-preview\lib\netstandard2.0\Microsoft.Azure.Manage
ment.Network.dll" to "E:\az-github\azure-powershell\artifacts/Debug\Az.Network\Microsoft.Azure.Management.Network.dll".

- Could you give us a guide to update azure sdk for net?
- Any doc to define default set of parameter?

**Comment by niander** [comment]
@ninweninwe If your changes are in 19.12 version we are updating it in the network-april branch and you should be able to use it by merging it back to your branch.

File: src/Storage/Storage/Storage.csproj [hunk 1]
```diff
@@ -1,24 +1,24 @@
-<Project Sdk="Microsoft.NET.Sdk">
```
Comment (MiYanni) [inline src/Storage/Storage/Storage.csproj]: Why did you change the Storage.csproj file? Please revert these changes.

File: src/Network/Network/help/Get-AzVirtualNetworkGatewayVpnclientConnectionHealth.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+# Get-AzVirtualNetworkGatewayVpnclientConnectionHealth
+
+## SYNOPSIS
+Get the list of vpn client connection health of an Azure virtual network gateway for per vpn client connection
+
+## SYNTAX
+
+```
+Get-AzVirtualNetworkGatewayVpnclientConnectionHealth -VirtualNetworkGatewayName <String> -ResourceGroupName <String> -InputObject <PSVirtualNetworkGateway> -ResourceId <ResourceId>
```
Comment (MiYanni) [inline src/Network/Network/help/Get-AzVirtualNetworkGatewayVpnclientConnectionHealth.md]: It looks like documentation wasn't updated from when the name became Get-AzVirtualNetworkGatewayVpn**C**lientConnectionHealth. Could you regenerate it?

File: src/Network/Network/help/Az.Network.md [hunk 3]
```diff
@@ -464,6 +464,9 @@ Gets a virtual network tap
 ### [Get-AzVirtualNetworkUsageList](Get-AzVirtualNetworkUsageList.md)
 Gets virtual network current usage.
 
+### [Get-AzVirtualNetworkGatewayVpnclientConnectionHealth](Get-AzVirtualNetworkGatewayVpnclientConnectionHealth.md)
```
Comment (MiYanni) [inline src/Network/Network/help/Az.Network.md]: It looks like documentation wasn't updated from when the name became Get-AzVirtualNetworkGatewayVpn**C**lientConnectionHealth. Could you regenerate it?

File: src/Network/Network/help/Get-AzVirtualNetworkGatewayVpnclientConnectionHealth.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+egressBytesTransferred
+ingressPacketsTransferred
+ingressBytesTransferred
+maxPacketsPerSecond
+
+## EXAMPLES
+
+### Example 1
+```
+PS C:\> Get-AzVirtualNetworkGatewayVpnclientConnectionHealth -ResourceGroupName resourceGroup -VirtualNetworkGatewayName gatewayName
```
Comment (MiYanni) [inline src/Network/Network/help/Get-AzVirtualNetworkGatewayVpnclientConnectionHealth.md]: Example isn't using valid PowerShell. If `resourceGroup` and `gatewayName` are variables, they would be accessed via `$resourceGroup` and `$gatewayName`. Also, cmdlet name should be Get-AzVirtualNetworkGatewayVpn**C**lientConnectionHealth. And, the example output is not from PowerShell (it wouldn't display anything like that).

**Comment by MikhailTryakhov** [comment]
@number213 what's up here?

**Comment by number213** [comment]
Test run timed out

**Comment by number213** [comment]
New run had a failure during DataLakeStore's `TestAdlsFileSystem` which is unrelated to this PR. Will re-run CI again

File: src/Network/Network/Network.format.ps1xml [hunk 5]
```diff
@@ -647,6 +647,68 @@
         </ListEntries>
       </ListControl>
     </View>
+    <View>
+      <Name>Microsoft.Azure.Commands.Network.Models.PSVpnClientConnectionHealthDetail</Name>
```
Comment (MiYanni) [inline src/Network/Network/Network.format.ps1xml]: This view is in List format. It should be in Table format.

File: src/Network/Network/Models/PSVpnClientConnectionHealthDetail.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.WindowsAzure.Commands.Common.Attributes;
+
+namespace Microsoft.Azure.Commands.Network.Models
+{
+    public partial class PSVpnClientConnectionHealthDetail
+    {
+        [Ps1Xml(Label = "VpnConnectionId", Target = ViewControl.Table)]
```
Comment (MiYanni) [inline src/Network/Network/Models/PSVpnClientConnectionHealthDetail.cs]: For these attributes to be useful, you need to generate the format.ps1xml from the models. See documentation here on how to do that: https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/generate-format.ps1xml-file.md#how-to-generate-formatps1xml-file

**Comment by MikhailTryakhov** [comment]
@ninweninwe can you please help with changelog latest improvement to make it merged

**Comment by ninweninwe** [comment]
> @ninweninwe can you please help with changelog latest improvement to make it merged

I have updated changelog.md with Get-AzVirtualNetworkGatewayVpnClientConnectionHealth.  What else do I need to do with the latest improvement?
